src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -154,202 +110,179 @@ public HttpConnection(Uri url, ITransportFactory transportFactory, ILoggerFactor
         }
 
         public Task StartAsync() => StartAsync(TransferFormat.Binary);
-        public async Task StartAsync(TransferFormat transferFormat) => await StartAsyncCore(transferFormat).ForceAsync();
 
-        private Task StartAsyncCore(TransferFormat transferFormat)
+        public async Task StartAsync(TransferFormat transferFormat)
+        {
+            await StartAsyncCore(transferFormat).ForceAsync();
+        }
+
+        private async Task StartAsyncCore(TransferFormat transferFormat)
         {
-            if (ChangeState(from: ConnectionState.Disconnected, to: ConnectionState.Connecting) != ConnectionState.Disconnected)
+            CheckDisposed();
+
+            if (_started)
             {
-                return Task.FromException(
-                    new InvalidOperationException($"Cannot start a connection that is not in the {nameof(ConnectionState.Disconnected)} state."));
+                return;
```

Comment:
Log SkippingStart?

Reply:
I still think this should wait on the previous call.

Reply:
Yep, we talked about that, I'll put it in. I did it in HubConnection already

Reply:
Actually, it already does. Since `_started` is set at the end, then it will only be true if the connection is fully started. If the second call passes this check, it takes the lock which waits on the first call to finish. Once it's finished, `_started` is `true` and it no-ops.